lib/srv/ctx.go

```diff
@@ -369,6 +369,9 @@ func NewServerContext(ctx context.Context, parent *sshutils.ConnectionContext, s
 		cancel:                 cancel,
 	}
 
+	// Pre-setup log entry in case early failures need to log something.
+	child.initLogEntry()
```

Comment:
Suggestion: after looking at this a bit I think I would go for a somewhat simpler Entry initialization right where we initialize the ServerContext (line 370 above) and rename `initLogEntry` to `updateLogEntry`, so we 1) get a completely initialized instance right from the start and 2) it looks a bit less suspicious.

That said, feel free to push back.

Reply:
=D i was thinking to initialize as many fields as we can in the first try. but it does look a bit suspicious.

Updated logic.

Thanks